### DISCUSSION TITLE
How to create custom node with custom modal / pop-up?

### DISCUSSION THREAD

**Discussion by thipokKub** [Comfy-Org/ComfyUI#5002] [General]
ComfyUI is very useful especially the Mask Editor. It open up a modal window, before returning to the normal workflow. If I want to do something like that on my custom node, can I do that as well?

**Comment by ltdrdata** [comment]
https://github.com/comfyanonymous/ComfyUI/blob/3d4d1ae4aaea07aa7c25fc72ad1bd2b21ccfdbeb/web/extensions/core/maskeditor.js


This is the implementation used before the new TypeScript-based front-end. Since it can function independently as a .js file, it should be easier to understand

(And this is the new implementation
https://github.com/Comfy-Org/ComfyUI_frontend/blob/main/src/extensions/core/maskeditor.ts )

https://github.com/ltdrdata/ComfyUI-Inspire-Pack/blob/c0d54d8a88c0b900998cc23f4041508d258b388d/__init__.py#L39

To create a web extension, you need to specify the WEB_DIRECTORY like this.

https://github.com/pythongosssss/ComfyUI-Custom-Scripts

And when creating a web extension, referring to this repository will be very educational
